azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -249,6 +277,25 @@ public Project createProject(String projectName, String description,
     return newProject;
   }
 
+    /**
+     * Permanently delete all project files and properties data for all versions
+     * of a project and log event in project_events table
+     *
+     * @param project
+     * @param deleter
+     * @return
+     * @throws ProjectManagerException
+     */
+    public synchronized Project purgeProject(Project project, User deleter)
+        throws ProjectManagerException {
+        projectLoader.cleanOlderProjectVersion(project.getId(),
```

Comment:
I expect to throw exception in that scenario. For purge API, I have handled that case while invoking this method.